### PR TITLE
CLS2-1033: Add EYB lead source to Company Activity model

### PR DESCRIPTION
**Pull request by samuele-mattiuzzo**
### Description of change

This PR addresses [CLS2-1033](https://uktrade.atlassian.net/browse/CLS2-1033)

It's a small chunk of a bigger piece of work that aims to add EYB Lead to the Company Activity tab and simply sets the scene for it to happen:

- [x] Updates the Company Activity model by linking EYB as foreign key and adding new source choice (EYBLead).
- [x] Creates `CompanyActivityEYBLeadFactory`

Sync and search will be dealt with in two separate PRs

### Checklist

* [x] Has this branch been rebased on top of the current `main` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/main/docs/CONTRIBUTING.md) for more guidelines.

[CLS2-1033]: https://uktrade.atlassian.net/browse/CLS2-1033?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: datahub/company_activity/tests/factories.py [hunk 1]
```diff
@@ -220,3 +221,25 @@ def _create(cls, model_class, *args, **kwargs):
...
+    """
+
+    activity_source = CompanyActivity.ActivitySource.eyb_lead
+    eyb_lead = factory.SubFactory(EYBLeadFactory)
+
+    class Meta:
+        model = 'company_activity.CompanyActivity'
+
+    @classmethod
+    def _create(cls, model_class, *args, **kwargs):
```
Comment (samuele-mattiuzzo) [inline datahub/company_activity/tests/factories.py:238]: this is currently not tested as it will be implemented in a separate PR

**Review by baarkerlounger — APPROVED** [review]

**Comment by samuele-mattiuzzo** [comment]
> Think you need to rebase your migrations

yeah just noticed, did a merge offline and obviously this wasn't raised. on it 👍🏻

**Review by cgsunkel — APPROVED** [review]

File: datahub/company_activity/migrations/0013_companyactivity_eyb_lead_and_more.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    dependencies = [
+        ('investment_lead', '0009_alter_eyblead_marketing_hashed_uuid'),
+        ('company_activity', '0012_companyactivity_great_and_more'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='companyactivity',
+            name='eyb_lead',
+            field=models.ForeignKey(blank=True, help_text='If related to an EYB leady, must not have relations to any other activity (referral, event etc)', null=True, on_delete=django.db.models.deletion.CASCADE, related_name='activity', to='investment_lead.eyblead', unique=True),
```
Comment (swenban) [inline datahub/company_activity/migrations/0013_companyactivity_eyb_lead_and_more.py:18]: There's a little typo- 'leady' should be 'lead

Also, is the CASCADE delete option the best for this? It might put a lot of data at risk

File: datahub/company_activity/models/company_activity.py [hunk 3]
```diff
@@ -114,6 +115,19 @@ class ActivitySource(models.TextChoices):
...
 
+    eyb_lead = models.ForeignKey(
+        'investment_lead.EYBLead',
+        unique=True,
+        null=True,
+        blank=True,
+        related_name='activity',
+        on_delete=models.CASCADE,
+        help_text=(
+            'If related to an EYB leady, must not have relations to any other activity '
```
Comment (swenban) [inline datahub/company_activity/models/company_activity.py:126]: Same typo
